aws/validators.go

```diff
@@ -1972,3 +1972,20 @@ func validateDxConnectionBandWidth(v interface{}, k string) (ws []string, errors
 	errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validBandWidth, val))
 	return
 }
+
+func validateKmsKey(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	arnPrefixPattern := `arn:[\w-]+:([a-zA-Z0-9\-])+:([a-z]{2}-(gov-)?[a-z]+-\d{1})?:(\d{12})?:`
```

Comment:
Can we switch this to not assume anything about the format of AWS partitions, regions, and account IDs in the future please? `arn:[^:]+:kms:[^:]+:[^:]+:`